File: src/main/java/me/kotlia/KanjiComponent.java
package me.kotlia;

public class KanjiComponent {
    private String component;
    public KanjiComponent(String component) {
        this.component = component;
    }
    public String format() {
        String temp = this.component;
        Integer counter = null;
        temp = temp.substring(1, temp.length()-1);
        String[] eachArr = temp.trim().split("");

        int index = 0;
        for (String s : eachArr) {
            if (counter == null) {
                counter = 0;
            } else if (s.equals("[")) {
                counter = counter + 1;
            } else if (s.equals("]")) {
                counter = counter - 1;
            } else if (counter == 0) {
                temp = temp.substring(0, index-1) + ";" + temp.substring(index);
            }
            index++;
        }
        String[] arr = temp.split(";");
        StringBuilder res = new StringBuilder();
        String inner = "";
        for (String it : arr) {
            inner = it.substring(it.indexOf(",") + 2, it.length() - 1);
            res.append(inner.split(",")[0].replace("\"", ""));
        }
        return res.toString();
    }
}

File: src/main/kotlin/me/kotlia/KanjiComponent.kt
package me.kotlia

import java.net.URI
import java.net.URLEncoder
import java.net.http.HttpClient
import java.net.http.HttpRequest
import java.net.http.HttpResponse
import java.nio.charset.StandardCharsets
import com.google.gson.JsonParser

import com.google.gson.JsonObject
import java.lang.Exception
import javax.script.ScriptEngineManager


class KanjiComponent (private var component: String = "") {
    override fun toString(): String {
        return this.component
    }
    fun format(): String {
        var temp = this.component
        var counter: Int? = null
        temp = temp.substring(1, temp.length-1)
        temp.trim().split("").forEachIndexed { index, s ->
            if (counter == null) {
                counter = 0
            } else if (s == "[") {
                counter = counter!! + 1
            } else if (s == "]") {
                counter = counter!! - 1
            } else if (counter == 0) {
                temp = try {
                    temp.substring(0, index-1) + ";" + temp.substring(index)
                } catch (e: Exception) {
                    temp
                }
            }
        }
        val arr = temp.split(";")
        var res = ""
        var inner = ""
        arr.forEach {
            inner = it.substring(it.indexOf(",") + 2, it.length - 1)
            res += inner.split(",")[0].replace("\"", "")
        }
        return res
    }
}